test/fixtures/rules/std_rule_cases/AM08.yml

```diff
@@ -0,0 +1,140 @@
+rule: AM08
+
+test_pass_select_comma:
+  # this syntax is currently not covered by AM08
+  pass_str: "SELECT a.*, b.* FROM a, b"
+
+test_fail_missing_clause:
+  fail_str: "SELECT foo.a, bar.b FROM foo INNER JOIN bar;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo CROSS JOIN bar;\n"
+
+test_fail_missing_clause_lowercase_default:
+  fail_str: "SELECT foo.a, bar.b FROM foo left join bar;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo cross join bar;\n"
+
+test_fail_missing_multiple_joins:
+  fail_str: "SELECT foo.a, bar.b FROM foo left join bar on 1=2 left join baz;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo left join bar on 1=2 cross join baz;\n"
+
+test_fail_missing_multiple_joins_mixed_types:
+  fail_str: "SELECT foo.a, bar.b FROM foo left join bar on foo.x = bar.y left join baz;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo left join bar on foo.x = bar.y cross join baz;\n"
+
+test_fail_missing_multiple_joins_not_last:
+  fail_str: "SELECT foo.a, bar.b FROM foo left join bar left join baz on foo.x = bar.y;\n"
+  fix_str: "SELECT foo.a, bar.b FROM foo cross join bar left join baz on foo.x = bar.y;\n"
+
+test_fail_cross_join_with_where_clause:
+  fail_str: "SELECT foo.a, bar.b FROM foo JOIN bar WHERE foo.a = bar.a;\n"
+  violations:
+    - code: AM08
+      description: WHERE clause used for join condition. Use explicit ON instead.
+      end_file_pos: 57
+      end_line_no: 1
+      end_line_pos: 58
+      fixes: []
+      name: ambiguous.join_condition
+      start_file_pos: 38
+      start_line_no: 1
+      start_line_pos: 39
+      warning: false
```

Comment:
Where clause can contain combination of effective on clause and other where conditions. It can also be arbitrary nested expressions.

Decomposition is certainly possible for subset of cases, but we need to be careful not to introduce behavior change via autofix.

If we deem that autofix is critical, I'd be grateful for support in defining test case list.

Reply:
I agree about making this a separate rule. When rules have a fairly narrow scope, they are easier to document, understand, and maintain. If it would be helpful, the two rules could potentially share/reuse code -- a number of the existing rules do that.